### PR TITLE
[WIP] e2e metal upi steps

### PR DESCRIPTION
**Pull request by kirankt**
Testing and improving work authored-by: rna-afk here: #24085

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/metal/OWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+approvers:
+- staebler
+- patrickdillon
+- jhixson74
+- jstuever
+- rna-afk
```
Comment (patrickdillon) [inline ci-operator/step-registry/openshift/e2e/metal/OWNERS:6]: For all of these owners files can we remove @staebler, and add @kirankt @sadasu & @barbacbd

File: ci-operator/step-registry/upi/install/metal/upi-install-metal-commands.sh [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+if [[ "${CLUSTER_VARIANT}" =~ "fips" ]]; then
+  cat >> ${SHARED_DIR}/installer/install-config.yaml << EOF
+fips: true
+EOF
+fi
+
+cat >> ${SHARED_DIR}/installer/install-config.yaml << EOF
+pullSecret: >
+  ${PULL_SECRET}
+EOF
```
Comment (patrickdillon) [inline ci-operator/step-registry/upi/install/metal/upi-install-metal-commands.sh:112]: Can the install config go in its own pre step?

File: ci-operator/step-registry/upi/install/metal/upi-install-metal-commands.sh [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+  oc adm release mirror --from ${MIRROR_BASE}-scratch:release --to ${MIRROR_BASE} --to-release-image ${MIRROR_BASE}:mirrored
+  RELEASE_PAYLOAD_IMAGE_SHA=$(oc get istag ${MIRROR_BASE##*/}:mirrored -o=jsonpath="{.image.metadata.name}")
+  oc delete imagestream "$(basename "${MIRROR_BASE}-scratch")"
+  RELEASE_IMAGE_MIRROR="${MIRROR_BASE}@${RELEASE_PAYLOAD_IMAGE_SHA}"
+
+  echo "Installing from mirror override release ${RELEASE_IMAGE_MIRROR}"
+  OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="${RELEASE_IMAGE_MIRROR}"
+else
+  echo "Installing from release ${RELEASE_IMAGE_LATEST}"
+fi
```
Comment (patrickdillon) [inline ci-operator/step-registry/upi/install/metal/upi-install-metal-commands.sh:43]: I assume we're not using a mirror variant? We can probably just remove this then.

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml [hunk 4]
```diff
@@ -479,6 +479,14 @@ tests:
   steps:
     cluster_profile: ibmcloud
     workflow: openshift-e2e-ibmcloud
+- as: e2e-metal-upi
+  optional: true
+  skip_if_only_changed: (^(docs|upi|hack)/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Z]+\.md$)|(^(data/.*|pkg/.*|terraform/providers)/(alibabacloud|aws|azure|azurestack|baremetal|gcp|libvirt|nutanix|openstack|ovirt|powervs|vsphere)/)
```
Comment (patrickdillon) [inline ci-operator/config/openshift/installer/openshift-installer-master.yaml:484]: We don't want this job to run very often. I think
```suggestion
  run_if_changed: metal
```
would make it run for changes to baremetal or upi/metal, which seems good to me.

**Comment by kirankt** [comment]
Work will continue in #24085

**Pull request closed without merging**